diff --git a/Makefile b/Makefile
index 83758b2..af0f5b8 100644
--- a/Makefile
+++ b/Makefile
@@ -5,15 +5,16 @@
 #
 #

+include ../../common.mk
+
 CC = g++
 JAVAC = $(JAVA_HOME)/bin/javac
 JAVA = $(JAVA_HOME)/bin/java

 JNI_INCLUDES = $(JAVA_HOME)/include $(JAVA_HOME)/include/linux
-
-CFLAGS = -O3 -DNDEBUG -fPIC
+override CFLAGS = -I$(PMDK_EXAMPLES_INCLUDE) -O3 -DNDEBUG -fPIC
 JAVAFLAGS = -Xlint:unchecked -proc:none -XDenableSunApiLintControl
-LINK_FLAGS = -fPIC -O3 -DNDEBUG -shared -lpmem -lpmemobj -Wl,-rpath,/usr/local/lib:/usr/local/lib64
+LINK_FLAGS = -fPIC -O3 -DNDEBUG -shared -L$(PMDK_EXAMPLES_LIB_NONDEBUG) -lpmem -lpmemobj -Wl,-rpath,/usr/local/lib:/usr/local/lib64

 CPP_SOURCE_DIR = src/main/cpp
 JAVA_SOURCE_DIR = src/main/java
